src/java/voldemort/server/VoldemortConfig.java

```diff
@@ -148,6 +148,7 @@
     public static final String PUSH_HA_LOCK_PATH = "push.ha.lock.path";
     public static final String PUSH_HA_LOCK_IMPLEMENTATION = "push.ha.lock.implementation";
     public static final String PUSH_HA_MAX_NODE_FAILURES = "push.ha.max.node.failure";
+    public static final String PUSH_HA_STATE_AUTO_CLEANUP = "push.ha.state.auto.cleanup";
```

Comment:
I think it would be better to prove that deleting the state is the right thing to do, than to put the decision behind a configuration variable.  That's just delegating the decision to an SRE who has less time and context to work out the right thing to do, and already has a manual workaround.